### DISCUSSION TITLE
CMYK and Pantone color separations of an image

### DISCUSSION THREAD

**Discussion by PatrickZelos** [ImageMagick/ImageMagick#6524] [Help]
### ImageMagick version

7.1.1-15

### Operating system, version and so on

Windows 10

### Description

Hi there,

I want to extract multiple defined color planes from pdf or tiff files into a grayscale TIFFS (one for each color in the file)

So far I found that it might be possible to do CMYK separation, but is it also possible to extract defined pantone colors separately?

Regards, Patrick

**Comment by snibgo** [comment]
In a TIFF or PDF, the CMYK channels are usually recorded in the file as channels of an image. IM can read and separate them into individual files. Pantone colors are often recorded as "spot" colors that IM cannot read.

If you link to samples, or zip them and upload them here, we can take a look.

**Reply by PatrickZelos** [reply]
Thanks for this information. I attached a small Test PDF file It contains CYMK (Cyan plane is empty) and Pantone 175 C and Pantone 727 C. Please take a look.
Thanks!
[Test.pdf](https://github.com/ImageMagick/ImageMagick/files/12226046/Test.pdf)

**Comment by snibgo** [comment]
`test.pdf` is a vector CMYK file. The image is 0.24 x 1.97 inches (according to Adobe Acrobat Reader).

ImageMagick rasterizes the vector image (turns it into pixels) by delegating that task to Ghostscript. We can create any number of pixels by using "`-density 300`" or whatever number we want.

We can then separate the CMYK channels into separate grayscale raster files.
```
magick -density 300 test.pdf -separate x_%02d.png
```
That creates three files: `x_00.png`, `x_01.png` etc, for C, M, Y and K.

IM cannot see any Pantone colours. Nor can exiftool.

>  (Cyan plane is empty)

I don't know what that means. The Cyan channel has non-zero values.

**Comment by PatrickZelos** [comment]
I created the file with Corel Draw. 
![image](https://github.com/ImageMagick/ImageMagick/assets/141017119/dff9cd2e-3fb7-4bde-8c82-e8e7a0ae8dc1)
Each of the circles has one of the colors.
1 Magenta
2 Yellow
3 Pantone 727 C
4 Pantone 175 C
5 Black

It sould be all 0 in the cyan color channel
The image size is 6x50 mm which is nearly your suggested size

**Comment by snibgo** [comment]
I suppose some software has converted those Pantone colours to CMYK. Either Corel Draw has done this when creating the PDF, or Ghostscript when rasterizing the PDF.

The PDF contains an ICC color profile, but this is for CMYK channels only, and does not contain Pantone colours.

When you read the PDF file with Corel Draw, does the result have Pantone colours?

**Comment by PatrickZelos** [comment]
Yes, when I reimport the pdf file to Corel Draw, it still has Pantone colors.

**Comment by snibgo** [comment]
That's good. Perhaps Ghostscript has a method to extract the Pantone channels.

**Reply by PatrickZelos** [reply]
OK, I will check this

**Comment by dceeee** [comment]
Ghostscript works well for pdf's
gswin64 -sDEVICE=tiffsep -dNOPAUSE -dBATCH -dSAFER -r300x300 -sOutputFile=Test.tif Test.pdf
produces the following files:
Test(Black).tif
Test(Cyan).tif
Test(Magenta).tif
Test(PANTONE 175 C).tif
Test(PANTONE 727 C).tif
Test(Yellow).tif
Test.tif

Test.tif is a CMYK composite, the others are the channels in grayscale

**Reply by PatrickZelos** [reply]
Thanks!

**Comment by snibgo** [comment]
Excellent, thanks @dceeee
